Plugins/Events/Events/CombatEvents.cpp

```diff
@@ -42,4 +63,56 @@ void CombatEvents::StartCombatRoundHook(
     Events::SignalEvent(before ? "NWNX_ON_START_COMBAT_ROUND_BEFORE" : "NWNX_ON_START_COMBAT_ROUND_AFTER" , thisPtr->m_pBaseCreature->m_idSelf);
 }
 
+void CombatEvents::AttackHook(
+    Hooks::CallType type,
+    API::CNWSCombatRound* thisPtr,
+    unsigned char attackNumber)
+{
+    if ( type == Hooks::CallType::AFTER_ORIGINAL )
+    {
+        // SetCurrentAttack is 1-based, GetAttack is 0-based
+        CNWSCombatAttackData* combatAttackData = g_combatEvents->m_combatAttackData = thisPtr->GetAttack(attackNumber - 1);
+        Events::PushEventData("TARGET_OBJECT_ID", Utils::ObjectIDToString(thisPtr->m_pBaseCreature->m_oidAttackTarget));
+        Events::PushEventData("ATTACK_NUMBER", std::to_string(static_cast<int>(attackNumber)));
+        for ( int i = 0; i < 13; i++ )
+        {
+            int16_t nDamage = combatAttackData->m_nDamage[i];
+            Events::PushEventData("ATTACK_DAMAGE_" + std::to_string(i), std::to_string(nDamage));
+        }
+        Events::PushEventData("ATTACK_RESULT", std::to_string(combatAttackData->m_nAttackResult));
+        Events::PushEventData("IS_SNEAK_ATTACK", std::to_string(combatAttackData->m_bSneakAttack || combatAttackData->m_bDeathAttack));
+        Events::PushEventData("IS_OFFHAND_ATTACK", std::to_string(combatAttackData->m_nWeaponAttackType == 2));
+        Events::SignalEvent("NWNX_ON_ATTACK", thisPtr->m_pBaseCreature->m_idSelf);
+        g_combatEvents->m_combatAttackData = nullptr;
+    }
+}
+
+ArgumentStack CombatEvents::AddAttackDamage(ArgumentStack&& args)
```

Comment:
This is better done with `SetEventResult()`, see [here](https://github.com/nwnxee/unified/blob/master/Plugins/Events/Events/StealthEvents.cpp#L50-L76), [here](https://github.com/nwnxee/unified/blob/master/Plugins/Events/Events/ClientEvents.cpp#L55-L84) and [here](https://github.com/nwnxee/unified/blob/master/Plugins/Events/Events/ItemEvents.cpp#L231-L242) for examples.

Reply:
The problem is that the result isn't a single integer, but an array of integers (one per damage type). So at a minimum we'd need to worry about serialization - are there functions readily available for this (similar to argument/return serialization for function calls)?

Reply:
Simplified the functionality to only deal with base damage. More restrictive but much simpler (using SetEventResult now), and probably all that's really needed (other damage type modifications can be handled via the Damage plugin).